### ISSUE TITLE
[FEAT] Compile all content into one language json file from three separate files

### ISSUE THREAD

**Issue by ngiangre** [COVID-19-electronic-health-system/Corona-tracker#643]
# ⚠️ IMPORTANT: Please fill out this template to give us as much information as possible to consider/implement the feature.

### Prerequisites

- [ ] check this box if you have completed the following:
    - Reviewed the [contributing guidelines](https://github.com/COVID-19-electronic-health-system/.github/blob/master/CONTRIBUTING.md) and [support](https://github.com/COVID-19-electronic-health-system/.github/blob/master/SUPPORT.md) files
    - Reviewed the README file for the repository you are working in
    - Searched for relevant instructions [on our Discord server](https://discord.gg/pPERUuv)
    - Searched the issues of the repository you are working in to make sure one was not already filed

### Summary

Currently the education, health, and display text content for each language are three separate json files. For the front end end team to work with a particular language they need to have each language be one json file that's referenced in the locale chosen. 

### Motivation

This is to promote reference by the app to the proper, central language json file

### Possible Alternatives

Keeping as is, which is a fragmented approach

### Additional Context

This is part of the content continuous integration workflow being developed

**Comment by BrianHHough** [comment]
Hi @ngiangre JSON file is updated here: https://github.com/COVID-19-electronic-health-system/Corona-tracker/blob/master/Translations/appText.json

cc: @salvolpe @tesla809 @SomeMoosery

**Comment by ngiangre** [comment]
Extremely detailed @BrianHHough! We can definitely make use of this.

**Comment by salvolpe** [comment]
Seriously!! Awesome work @BrianHHough

**Comment by SomeMoosery** [comment]
Undid this change - there's a PR process these things need to go through and we can't commit right to master. Don't want to be dictatorial but this can seriously break things. Will need a PR relating to this issue just like everything else.

**Comment by ngiangre** [comment]
Yeah agreed thanks @SomeMoosery. But @salvolpe and I can work with this file if @BrianHHough sends it through discord.

**Comment by SomeMoosery** [comment]
Yeah no prob! The file looks great, I agree - just need to make sure we're going through the git flow so things don't start to fall off track.

I'm also not sure what you mean - what are you trying to update? If you're trying to update components when some value changes that should be done with hooks in the React code.

**Comment by ngiangre** [comment]
Yeah don't want to obstruct the process at all. 

We need to make sure the text is adequately translated dependent on the locale. We want to try to help you make your hooks with the proper content:
 
![CoronaTracker - content continuous integration](https://user-images.githubusercontent.com/7811531/80321447-0f2ea180-87eb-11ea-981f-859a85cd1c2b.png)

I should've first made an issue pertaining to this. @BrianHHough prematurely did a PR but it's ok you live and learn. 

I need to make a more formal issue and description, but I guess this will do.

The idea is to allow hooks to be made dependent on the locale chosen (a file with the adequate translation will be pointed to for the correct translation). Sorry I'm not sure how much I can explain here. @SomeMoosery what would be the best way to proceed from here?

**Comment by SomeMoosery** [comment]
Yeah, so it seems like what can be done from this end is that once we get a master doc for each language merged in, we can create a new issue for which we change the locations specified in i18next to point to these new JSON files, and we should be set 👍

**Comment by ngiangre** [comment]
Ok great. I made a PR that was approved which put language json files in `docs/content` but now I'm working on another that puts everything in that language from 3 to one json file with the structure needed by front end - I'm working with @pavel-ilin and @AdhamAH on that. But @SomeMoosery if you have any suggestion on a better structure for the json files in `docs/content` let me know!

**Comment by ngiangre** [comment]
Closing other issues to have this be the issue with all the information.

https://github.com/COVID-19-electronic-health-system/Corona-tracker/issues/623#issue-605061240

https://github.com/COVID-19-electronic-health-system/Corona-tracker/issues/359#issue-592764712

https://github.com/COVID-19-electronic-health-system/Corona-tracker/issues/530#issue-601336342

**Comment by tesla809** [comment]
Can we make sure that the privacy policies and terms and conditions are also included as all content? If so, this can be within this issue.

**Comment by ngiangre** [comment]
Yes @tesla809

**Comment by AdhamAH** [comment]
@BrianHHough 
I loved the file! It is clear and will be easy to use i18next 
@ngiangre will you use this? or the examples you sent me? 
As @SomeMoosery said as soon as the JSON ready we can start working on point the app to use i18next when needed

**Comment by ngiangre** [comment]
I'm working on it.

The issue is that we need to have the structure in our data model so However we make the keys is consistent, makes sense, and the values can be adapted in the future. 
The data model is here:
https://docs.google.com/spreadsheets/d/1gifZGnj-ntIOXbZrRY4S5KEULoV64DCbi0pZSVWKsns/edit?usp=sharing

Right now only the health sheet follows the data model and has a field name to be included in the tag:
https://docs.google.com/spreadsheets/d/1tjeTH99QZSmyfG-TvX5seSYKa03yAHdMWrZt1z9521I/edit#gid=0

We need to fill in the parent/child keys and add a field name for education:
https://docs.google.com/spreadsheets/d/1OdasOk5y8w9JLv3VWmziV8XkznaYiW-OQzbQq5ZK26M/edit?usp=sharing

And Display Text is incomplete
https://docs.google.com/spreadsheets/d/1wn02oIks4avznMoJfxVHCo5221D9wwfu6yaWpi6QleA/edit?usp=sharing

I need help with the data model - I can handle this but @salvolpe can help and anyone else willing to help please reach out. And once the data model is done I can get the content from translations to populate the value via the python script `src/python/pull_gsheets_data.py`.

I work a lot with common data models so this is really important.

**Comment by pavel-ilin** [comment]
@BrianHHough I love your `appText.json`.

I would suggest modification of the survey section. I would like to store all questions in the array and add question type attribute to every question. This will help us not only with translation but also render all questions dynamically. We can iterate over the array and base on type of the question render components we need. This way we can move away from hardcoding every question.

Example:
![](https://i.ibb.co/XZNnD04/Screen-Shot-2020-04-27-at-2-57-37-PM.png)

I thin we can do the same with quizzes. 

@ngiangre @AdhamAH @salvolpe @tesla809 @SomeMoosery What do you think?

**Comment by ngiangre** [comment]
Great! The structure is do-able. Can you make a sample @pavel-ilin for the education content and display text as well?

**Comment by ngiangre** [comment]
@pavel-ilin @AdhamAH or anyone on front end - to finish converting Sal and I need to understand what exactly is hard coded and how values are set to defaults so we can accommodate in our data model. Can someone ping me, @NickG, on discord or comment here?

**Comment by AdhamAH** [comment]
I really still don't fully understand what are you trying to do and why it looks so complicated @ngiangre 
Anyway I hope this would help you 
https://docs.google.com/document/d/1f8RQFBNfLKEPPp2v1eYf693zU9hK8oEKq8Lb0S2VbFI/edit?usp=sharing

**Comment by ngiangre** [comment]
This is great @AdhamAH! Let’s chat on discord so we can be on the same page.

**Comment by SomeMoosery** [comment]
I think just to keep things clear in terms of this issue. The end result here is that we just end up with "master" JSON files for each language, which will be stored in `/public/locales`. We use i18next's `useTranlsation` hook, which pulls the currently-selected language from this directory and populates it by using the `t(...)` function.

There is no hard and fast rule on what's hard-coded and what's not, as every component is different just by nature of building out a responsive web app. This simply requires someone going through each component and looking for where there is hardcoded text, and making sure that text is captured in the master JSON file for translations.

I saw Brian was breaking his WIP master JSON file `appText.json` in terms of components, which I think is a good idea. For example, `onboardSection: { ... }` would correspond to `Onboarding.jsx`, `disclaimerSection: { ... }` would correspond to `Disclaimer.jsx` - so on and so forth.

This will be tedious but having someone go through and report back where things are hardcoded and need translating wouldn't save any time, as someone then still has to go through and add those hardcoded portions into the master JSON file, and use the `useTranslation` hook's `t(...)` function to translate.

This will likely be an iterative process - we can always create more issues to reformat the master JSON files and subsequently update the translated values in the corresponding components.

**Comment by ngiangre** [comment]
Correct @SomeMoosery - one master json for each language that will be within `public/locales/`

My goal is to get a "MVP" available - we will have every field needed in the gsheets that can then be transformed into the correct json file via python scripting with the google api.

In order to get the right structure, we needed to know how you would call the tags which wasn't clear. Now it is so thanks! You're right we might not be able to get everything, but we want to get most atleast. The over-engineered part might be from detailing everything, so we'll not do that. But we did need to understand the right structure so it's usable. The data model is important so it's consistent - if not in a data model, then iterative processing/modification/adding languages will get really convoluted and 'makeshift' real fast.

So I'm glad we are having this discussion!

**Comment by SomeMoosery** [comment]
That's a very good point!

Sounds good, since I've been somewhat away for a day or two I was just hoping to add some clarity as somewhat of an "outside" perspective to this translations bit. Seems like it's on the right track though thanks to yours and everyone's work on it, and this is one of the best discussion on an issue I think we've had 😄

**Comment by ngiangre** [comment]
teamwork makes the dream work!

**Comment by ngiangre** [comment]
Done. Finishing work in #665
